test/blackbox/run_blackbox_tests/test_cases_submesh.c

```diff
@@ -0,0 +1,296 @@
+                                    };
+static const char *node_ids[] = { "corenode1", "app1node1", "app2node1", "corenode2",
+                                  "app1node2", "app2node2"
+                                };
+
+static meshlink_node_status_lookup_t core_node1[] = { {INIT_ST, NODE_STARTED},
+	{NODE_STARTED, CHANNEL_OPENED},
+	{CHANNEL_OPENED, CHANNEL_DATA_RECIEVED}
+};
+
+static meshlink_node_status_lookup_t core_node2[] = { {INIT_ST, NODE_STARTED},
+	{NODE_STARTED, NODE_JOINED},
+	{NODE_JOINED, CHANNEL_OPENED},
+	{CHANNEL_OPENED, CHANNEL_DATA_RECIEVED}
+};
+
+static meshlink_node_status_lookup_t app1_node1[] = { {INIT_ST, NODE_STARTED},
+	{NODE_STARTED, NODE_JOINED},
+	{NODE_JOINED, CHANNEL_OPENED},
+	{CHANNEL_OPENED, CHANNEL_DATA_RECIEVED}
+};
+
+static meshlink_node_status_lookup_t app2_node1[] = { {INIT_ST, NODE_STARTED},
+	{NODE_STARTED, NODE_JOINED},
+	{NODE_JOINED, CHANNEL_OPENED},
+	{CHANNEL_OPENED, CHANNEL_DATA_RECIEVED}
+};
+
+static meshlink_node_status_lookup_t app1_node2[] = { {INIT_ST, NODE_STARTED},
+	{NODE_STARTED, NODE_JOINED},
+	{NODE_JOINED, CHANNEL_OPENED},
+	{CHANNEL_OPENED, CHANNEL_DATA_RECIEVED},
+	{CHANNEL_DATA_RECIEVED, CHANNEL_OPENED},
+	{CHANNEL_OPENED, CHANNEL_DATA_RECIEVED},
+	{CHANNEL_DATA_RECIEVED, MESH_EVENT_COMPLETED}
+};
+
+static meshlink_node_status_lookup_t app2_node2[] = { {INIT_ST, NODE_STARTED},
+	{NODE_STARTED, NODE_JOINED},
+	{NODE_JOINED, CHANNEL_OPENED},
+	{CHANNEL_OPENED, CHANNEL_DATA_RECIEVED},
+	{CHANNEL_DATA_RECIEVED, CHANNEL_OPENED},
+	{CHANNEL_OPENED, CHANNEL_DATA_RECIEVED},
+	{CHANNEL_DATA_RECIEVED, MESH_EVENT_COMPLETED}
+};
+
+/* State structure for SubMesh Test Case #1 */
+static char *test_case_submesh_1_nodes[] = { "corenode1", "app1node1", "app2node1", "corenode2", "app1node2", "app2node2" };
+static black_box_state_t test_case_submesh_1_state = {
+	.test_case_name =  "test_cases_submesh",
+	.node_names =  test_case_submesh_1_nodes,
+	.num_nodes =  6
+};
+
+static int black_box_group0_setup(void **state) {
+	const char *nodes[] = { "corenode1", "app1node1", "app2node1", "corenode2", "app1node2", "app2node2" };
+	int num_nodes = sizeof(nodes) / sizeof(nodes[0]);
+
+	PRINT_TEST_CASE_MSG("Creating Containers\n");
+	destroy_containers();
+	create_containers(nodes, num_nodes);
+
+	return 0;
+}
+
+static int black_box_group0_teardown(void **state) {
+	PRINT_TEST_CASE_MSG("Destroying Containers\n");
+	destroy_containers();
+
+	return 0;
+}
+
+static int black_box_all_nodes_setup(void **state) {
+	const char *nodes[] = { "corenode1" };
+	int num_nodes = sizeof(nodes) / sizeof(nodes[0]);
+
+	PRINT_TEST_CASE_MSG("Creating Containers\n");
+	destroy_containers();
+	create_containers(nodes, num_nodes);
+	PRINT_TEST_CASE_MSG("Created Containers\n");
+	return 0;
+}
+
+static bool change_state(meshlink_node_status_t *status, mesh_event_t currentEv) {
+	meshlink_node_status_lookup_t *lookup = status->lookup;
+
+	if(true == status->status) {
```

Comment:
Just write `if(status->status)`, the if-statement is by its very definition already checking if the argument is `true`.

Reply:
I have chnged the condition checking